libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -372,18 +371,42 @@ void CompilerUtils::convertType(Type const& _typeOnStack, Type const& _targetTyp
 			convertType(_typeOnStack, *_typeOnStack.mobileType(), true);
 		else if (targetTypeCategory == Type::Category::FixedPoint)
 		{
-			solAssert(
+			/*solAssert(
 				stackTypeCategory == Type::Category::Integer || 
 				stackTypeCategory == Type::Category::RationalNumber ||
 				stackTypeCategory == Type::Category::FixedPoint,
 				"Invalid conversion to FixedMxNType requested."
-			);
+			);*/
 			//shift all integer bits onto the left side of the fixed type
 			FixedPointType const& targetFixedPointType = dynamic_cast<FixedPointType const&>(_targetType);
-			if (auto typeOnStack = dynamic_cast<IntegerType const*>(&_typeOnStack))
-				if (targetFixedPointType.integerBits() > typeOnStack->numBits())
-					cleanHigherOrderBits(*typeOnStack);
-			solAssert(false, "Not yet implemented - FixedPointType.");
+			if (auto intType = dynamic_cast<IntegerType const*>(&_typeOnStack))
+ 			{
+				if (targetFixedPointType.integerBits() > intType->numBits())
+					cleanHigherOrderBits(_targetType/**intType*/);
+				u256 shiftFactor = u256(1) << (targetFixedPointType.fractionalBits());
+				m_context << shiftFactor << Instruction::MUL;
+			}
+			else if (auto rationalType = dynamic_cast<RationalNumberType const*>(&_typeOnStack))
+			{
+				bool isValidType = (
+					targetFixedPointType.integerBits() > rationalType->fixedPointType()->integerBits() || 
+					targetFixedPointType.fractionalBits() > rationalType->fixedPointType()->fractionalBits()
+				);
+				if (isValidType && _cleanupNeeded)
+					cleanHigherOrderBits(targetFixedPointType);
+				shiftFixedPointNumber(*rationalType->fixedPointType(), targetFixedPointType);
+			}
+			else 
+			{
+				FixedPointType const& stackFixedType = dynamic_cast<FixedPointType const&>(_typeOnStack);
+				if (
+					targetFixedPointType.integerBits() > stackFixedType.integerBits() || 
+					targetFixedPointType.fractionalBits() > stackFixedType.fractionalBits()
+				)
+					cleanHigherOrderBits(stackFixedType);
+				shiftFixedPointNumber(stackFixedType, targetFixedPointType);
+			}
```

Comment:
Please add some assertions about which types cannot end up here but are mentioned in the cases above.